2021-Firmware-Bootcamp/Core/Src/main.c

```diff
@@ -0,0 +1,211 @@
+  * @brief          : Main program body
+  ******************************************************************************
+  * @attention
+  *
+  * <h2><center>&copy; Copyright (c) 2023 STMicroelectronics.
+  * All rights reserved.</center></h2>
+  *
+  * This software component is licensed by ST under BSD 3-Clause license,
+  * the "License"; You may not use this file except in compliance with the
+  * License. You may obtain a copy of the License at:
+  *                        opensource.org/licenses/BSD-3-Clause
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+#include "spi.h"
+#include "tim.h"
+#include "usart.h"
+#include "gpio.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+static const int COUNTER_PERIOD_ = 60000;
+static const float DUTY_CYCLE_RANGE_ = 0.05;
+static const DUTY_CYCLE_MIN_ = 0.05;
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+
+  /* USER CODE BEGIN 2 */
+  GPIO_TypeDef * GPIOx = GPIOB;
+  uint16_t GPIO_Pin = GPIO_PIN_8;
+  GPIO_PinState PinState = GPIO_PIN_RESET;
+
+  HAL_GPIO_WritePin(GPIOx, GPIO_Pin, !PinState); //cs line put high in case it was low initially
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1); //timer started
+
+  uint8_t pTxData[3] = {0x01, 0x80, 0x00};
+  uint8_t pRxData[3] = {0};
+
+  int adcVal = 0;
```

Comment:
Usually we use datatypes that match the purpose of the variable. In this case, the compare register that you set later, `htim1->Instance->CCR1`, is a `uint32_t`, so we want to use that instead of `int`. Generally we don't use signed datatypes unless we have to.

If you're wondering how to find the that compare register, you can check the declaration of `__HAL_TIM_SET_COMPARE` and `htim1` via right-click + "Open Declaration".